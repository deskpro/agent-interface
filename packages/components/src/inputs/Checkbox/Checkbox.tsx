import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/elements/dp-inputs.css";

export type CheckboxProps = {
  label?: string | React.ReactNode;
  id?: string;
  undef?: boolean;
  className?: string;
  labelClassName?: string;
};

const Checkbox: React.SFC<
  CheckboxProps & React.HTMLProps<HTMLInputElement>
> = ({
  label = "",
  id = "checkbox",
  undef = false,
  className = "",
  labelClassName = "",
  ...props
}) => (
  <span
    className={classNames("dp-CustomCheckbox", {
      "dp-SelectedCheckbox": undef
    })}
  >
    <input
      type="checkbox"
      className={classNames("dp-ControlInput", className)}
      id={id}
      {...props}
    />
    <label htmlFor={id} className={labelClassName}>
      {label}
    </label>
  </span>
);

Checkbox.defaultProps = {
  label: ""
};

export default Checkbox;
