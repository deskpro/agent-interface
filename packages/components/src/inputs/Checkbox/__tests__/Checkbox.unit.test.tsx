import * as React from "react";
import { render } from "react-testing-library";

import Checkbox from "../Checkbox";

describe("<Checkbox />", () => {
  it("should match its snapshot", () => {
    const handleChange = jest.fn();
    const { container } = render(
      <Checkbox
        onChange={handleChange}
        label="Checkbox"
        checked
        labelClassName="label-class-name"
      />
    );
    expect(container.querySelector(".dp-CustomCheckbox")).toMatchSnapshot();
  });
});
