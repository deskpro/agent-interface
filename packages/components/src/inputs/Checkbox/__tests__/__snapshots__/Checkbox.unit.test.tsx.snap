// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> it should match its snapshot 1`] = `
<span
  class="dp-CustomCheckbox"
>
  <input
    checked=""
    class="dp-ControlInput"
    id="checkbox"
    type="checkbox"
  />
  <label
    class="label-class-name"
    for="checkbox"
  >
    Checkbox
  </label>
</span>
`;
